web/src/main/webapp/dataViews/tableView.ts

```diff
@@ -1100,6 +1113,28 @@ class HeavyHittersReceiver extends OnCompleteRenderer<TopList> {
         newPage.setDataView(hhv);
         this.page.insertAfterMe(newPage);
         hhv.fill(data.top, this.elapsedMilliseconds());
+        if (data.top.rows.length == 0)
+            this.showEmptyDialog();
+    }
+
+    protected showEmptyDialog(): void {
+        let min: number = 0.1;
+        let minString: string = min.toString() + "%";
+        let percentDialog = new Dialog("No Frequent Elements found", "");
+        percentDialog.addText("No Elements found with frequency above " + this.percent.toString() + "%.");
+        if (this.percent > min) {
+            percentDialog.addText("Lower the threshold? Can take any value above " + minString);
+            percentDialog.addTextField("newPercent", "Threshold (%)", FieldKind.Double, min.toString(),
```

Comment:
Do you also want to make this string a constant?
These constants belong in HeavyHittersView.